### PR TITLE
Snap 1024

### PR DESCRIPTION
**Pull request by rishitesh**
## Changes proposed in this pull request
a) In Spark 1.6 temp tables were stored against LogicalPlan of the DataFrame. Our catalog makes several assumptions regarding this to consider it as a temp table e.g. while dropping table.

However Spark2.0 analyses the logical plan and converts it to a LogicalRelation before storing that into catalog. Hence the assumptions break.

b) Excluded Univocity Parsers from snappy-core, as this library is already in compile time dependency of spark.sql

## Patch testing

Add a test. Running precheckin

## ReleaseNotes.txt changes

NA

## Other PRs 

NA

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -76,6 +76,7 @@ dependencies {
     exclude(group: 'org.scala-lang', module: 'scala-library')
     exclude(group: 'org.scala-lang', module: 'scala-reflect')
     exclude(group: 'org.scala-lang', module: 'scala-compiler')
+    exclude(group: 'com.univocity', module: 'univocity-parsers')
```
Comment (kneeraj) [inline core/build.gradle]: Was this causing any trouble or just by observation you found out?

File: core/src/main/scala/org/apache/spark/sql/SnappySession.scala [hunk 2]
```diff
@@ -907,15 +910,20 @@ class SnappySession(@transient private val sc: SparkContext,
           case _ => // ignore
         }
         cacheManager.uncacheQuery(Dataset.ofRows(this, plan))
-        sessionCatalog.unregisterDataSourceTable(tableIdent, Some(br))
+        if(sessionCatalog.isTemporaryTable(tableIdent)){ // This is due to temp table
```
Comment (kneeraj) [inline core/src/main/scala/org/apache/spark/sql/SnappySession.scala]: So "table not found" kind of error was coming when trying to drop temp tables which was directly trying unregisterDataSourceTable?

File: core/src/main/scala/org/apache/spark/sql/SnappySession.scala [hunk 3]
```diff
@@ -230,6 +230,9 @@ class SnappySession(@transient private val sc: SparkContext,
       case LogicalRelation(br, _, _) =>
         br match {
           case d: DestroyRelation => d.truncate()
+          case _ => if (!ignoreIfUnsupported) {
```
Comment (kneeraj) [inline core/src/main/scala/org/apache/spark/sql/SnappySession.scala]: Why is this required? How is the case of "_" related to temp tables only.

**Comment by kneeraj** [comment]
Looks ok to me except a couple of questions I have asked for my clarity.

File: core/build.gradle (hunk 1 above)
Comment (rishitesh) [inline core/build.gradle]: This causes trouble if we use CSV reader in our core/example projects, due to version conflicts

File: core/src/main/scala/org/apache/spark/sql/SnappySession.scala (hunk 2 above)
Comment (rishitesh) [inline core/src/main/scala/org/apache/spark/sql/SnappySession.scala]: Correct. As the plan is no more a LogicalPlan, no case was handling it.

File: core/src/main/scala/org/apache/spark/sql/SnappySession.scala (hunk 3 above)
Comment (rishitesh) [inline core/src/main/scala/org/apache/spark/sql/SnappySession.scala]: Its for cleanup of relations which supports truncate. It was causing a scala match error if we dont handle it.
